fiware-region-sanity-tests/tests/fiware_region_with_networks_tests.py

```diff
@@ -48,83 +50,110 @@ def __deploy_instance_helper__(self, instance_name, network_name=None, network_c
+            self.skipTest("Not all the networks were deleted")
+
+        # skip if not all routers were deleted
+        if self.test_world['routers']:
+            self.skipTest("Not all the routers were deleted")
+
         flavor_id = self.nova_operations.get_any_flavor_id()
         self.assertIsNotNone(flavor_id, "Problems retrieving a flavor")
 
         image_id = self.nova_operations.find_image_id_by_name(image_name=BASE_IMAGE_NAME)
         self.assertIsNotNone(image_id, "Problems retrieving the image '{}'".format(BASE_IMAGE_NAME))
 
-        network_id_list = None
-        if network_name is not None:
-            cidr = network_cidr if network_cidr is not None else "10.250.254.0/24"
-            network = self.neutron_operations.create_network_and_subnet(network_name, cidr=cidr)
-            self.test_world['networks'].append(network['id'])
-            network_id_list = [{'net-id': network['id']}]
-
-        if keypair_name is not None:
-            keypair_value = self.nova_operations.create_keypair(keypair_name)
-            self.test_world['keypair_names'].append(keypair_name)
-
-        security_group_name_list = None
-        if sec_group_name is not None:
-            sec_group_id = self.nova_operations.create_security_group_and_rules(sec_group_name)
-            self.test_world['sec_groups'].append(sec_group_id)
-            security_group_name_list = [sec_group_name]
-
-        server_data = self.nova_operations.launch_instance(instance_name=instance_name, flavor_id=flavor_id,
-                                                           image_id=image_id,
-                                                           network_id_list=network_id_list,
-                                                           metadata=metadata,
-                                                           keypair_name=keypair_name,
-                                                           security_group_name_list=security_group_name_list)
-        self.test_world['servers'].append(server_data['id'])
+        # instance prerequisites
+        try:
+            network_id_list = None
+            if network_name:
+                cidr = network_cidr or TEST_DEFAULT_CIDR
+                network = self.neutron_operations.create_network_and_subnet(network_name, cidr=cidr)
+                self.test_world['networks'].append(network['id'])
+                network_id_list = [{'net-id': network['id']}]
+
+            if keypair_name:
+                self.nova_operations.create_keypair(keypair_name)
+                self.test_world['keypair_names'].append(keypair_name)
+
+            security_group_name_list = None
+            if sec_group_name:
+                sec_group_id = self.nova_operations.create_security_group_and_rules(sec_group_name)
+                self.test_world['sec_groups'].append(sec_group_id)
+                security_group_name_list = [sec_group_name]
+        except (NovaClientException, NeutronClientException) as e:
```

Comment:
I suggest to have each try-except in order to identify exactly the problem when we see the message.